src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -143,12 +143,9 @@ public UnorderedCallAssertion MustHaveHappened(Repeated repeatConstraint)
 
             var asserter = this.asserterFactory.Invoke(this.Calls);
 
-            var description = new StringBuilderOutputWriter();
-            this.RuleBeingBuilt.WriteDescriptionOfValidCall(description);
+            asserter.AssertWasCalled(this.Matcher.Matches, this.RuleBeingBuilt.WriteDescriptionOfValidCall, repeatConstraint.Matches, repeatConstraint.ToString());

```

Comment:
Nitpick: couldn't we pass `repeatConstraint` directly instead of its string representation?